src/components/CharacterTab.jsx

```diff
@@ -222,6 +232,33 @@ export default function CharacterTab() {
     cellStyle: { display: 'flex' },
   }), [])
 
+  const externalFilterChanged = useCallback((newValue) => {
+    setCurrentFilters(newValue)
+    gridRef.current.api.onFilterChanged()
+  }, [])
+
+  const charinfo = JSON.parse(JSON.stringify(DB.getMetadata().characters))
+
+  const doesExternalFilterPass = useCallback(
+    (node) => {
+      if (currentFilters.element.length && !currentFilters.element.includes(charinfo[node.data.id].element)) {
+        return false
+      }
+      if (currentFilters.path.length && !currentFilters.path.includes(ClassToPath[charinfo[node.data.id].path])) {
+        return false
+      }
+      if (!((charinfo[node.data.id].name).toLowerCase().includes(currentFilters.name) || (charinfo[node.data.id].displayName).toLowerCase().includes(currentFilters.name))) {
+        return false
+      }
+      return true
```

Comment:
Can probably Demorgans law this into something simpler, remove the negation by returning the inverse

```
    return (filteredCharacter.name).toLowerCase().includes(currentFilters.name)
      || (filteredCharacter.displayName).toLowerCase().includes(currentFilters.name)
```